client/src/components/Search.js

```diff
@@ -3,35 +3,33 @@ import { useState } from "react";
 
 const Search = ({ setSearchQuery }) => {
 	const [searchInput, setSearchInput] = useState("");
-
+	const [isLoading, setIsLoading] = useState(false);
 	const handleChange = (event) => {
 		setSearchInput(event.target.value.trim());
 	};
 
 	const handleSubmit = async (event) => {
 		event.preventDefault();
+		setIsLoading(true);
 		setSearchQuery(searchInput);
+		setTimeout(() => {
+			setIsLoading(false);
+ }, 3000);
```

Comment:
This does work - I will be shown a spinner when I start to search - but it would be better if the `isLoading` state is set to false at the point when the api request is complete.

I think that we can merge this code as it is now. And then we can create a ticket to improve this afterwards to make it so that the spinner only shows while the api call is in progress.

I think a new ticket makes sense because it looks like we will need to change the state logic a bit so that we pass down a `isLoading` prop from the `Home` component, rather than having the state here. But lets come back to this.